### PR TITLE
feat: Enable `OR` Query support.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    Query query1 =
+        collection.where(
+            Filter.or(Filter.equalTo("a", 2), Filter.inArray("b", Arrays.asList(2, 3))));
+    checkQuerySnapshotContainsDocuments(query1, "doc3", "doc4", "doc6");
+
+    // a==2 || (b != 2 && b != 3)
+    // Has implicit "orderBy b"
+    Query query2 =
+        collection.where(
+            Filter.or(Filter.equalTo("a", 2), Filter.notInArray("b", Arrays.asList(2, 3))));
```
Comment (MarkDuckworth) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java:263]: This test doesn't match the comment on line 259. But the comment doesn't match the test name on 242.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+    // Query: a==1 || b==1
+    // There's no explicit nor implicit orderBy. Documents with missing 'a' or missing 'b' should be
+    // allowed if the document matches at least one disjunction term.
+    Query query5 = collection.where(Filter.or(Filter.equalTo("a", 1), Filter.equalTo("b", 1)));
+    checkQuerySnapshotContainsDocuments(query5, "doc1", "doc2", "doc4", "doc5");
+  }
+
+  @Test
+  public void orQueryWithInAndNotIn() throws Exception {
```
Comment (MarkDuckworth) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java:242]: This test is almost identical to the test on 268, except for 268 is plural. Just double checking that this was intentional.

**Review by MarkDuckworth — APPROVED** [review]
Approved with suggestions.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+@RunWith(JUnit4.class)
+public class ITQueryTest {
+
+  private static Firestore firestore;
+
+  @Rule public TestName testName = new TestName();
+
+  @Before
+  public void setUpFirestore() {
+    firestore = FirestoreOptions.newBuilder().setHost("localhost:8080").build().getService();
```
Comment (MarkDuckworth) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java:55]: Do these tests always run against localhost? Consider adding a TODO here?

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+            .limit(1)
+            .orderBy("a"),
+        "doc5");
+
+    // Test with limits (explicit order by DESC): (a==2) || (b == 1) ORDER BY a LIMIT_TO_LAST 1
+    checkQuerySnapshotContainsDocuments(
+        collection
+            .where(Filter.or(Filter.equalTo("a", 2), Filter.equalTo("b", 1)))
+            .limitToLast(1)
+            .orderBy("a"),
```
Comment (MarkDuckworth) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java:184]: orderBy is not DESC

Comment (ehsannas) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java:184]: The `limitToLast(1).orderBy("a")` is in fact `limit(1).orderBy("a", "desc")`

I will add another test to perform the latter too.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java (hunk 3 above)
Comment (ehsannas) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java:57]: Done.

**Pull request merged**